### ISSUE TITLE
denols: jumping to definition then returning breaks lsp

### ISSUE THREAD

**Issue by sigmaSd** [neovim/nvim-lspconfig#1998]
### Description

jumping to definition then returning breaks lsp
![denovim](https://user-images.githubusercontent.com/22427111/178674021-e7624ace-22f9-4773-9eec-a76b5d74d4f2.gif)

coc-deno used to have the same issue but it was recently fixed https://github.com/fannheyward/coc-deno/issues/139

### Neovim version

NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3

### Nvim-lspconfig version

41a826947eefbc7b59cc80cef405cce9ea6d82b8

### Operating system and version

Linux mrcool 5.18.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 02 Jul 2022 21:03:06 +0000 x86_64 GNU/Linux

### Affected language servers

denols

### Steps to reproduce

1.  `echo "console.log(4)" > test.ts`
2. `nvim test.ts`
3. move cursor to log and press `gd` to go to definition
4. press ctrl-o to return

### Actual behavior

Notice that lsp is not working correctly any more and have errors and the console and log have type any

### Expected behavior

the lsp should continue working correctly.

### Minimal config

```Lua
local on_windows = vim.loop.os_uname().version:match 'Windows'

local function join_paths(...)
    local path_sep = on_windows and '\\' or '/'
    local result = table.concat({ ... }, path_sep)
    return result
end

vim.cmd [[set runtimepath=$VIMRUNTIME]]

local temp_dir = vim.loop.os_getenv 'TEMP' or '/tmp'

vim.cmd('set packpath=' .. join_paths(temp_dir, 'nvim', 'site'))

local package_root = join_paths(temp_dir, 'nvim', 'site', 'pack')
local install_path = join_paths(package_root, 'packer', 'start', 'packer.nvim')
local compile_path = join_paths(install_path, 'plugin', 'packer_compiled.lua')

local function load_plugins()
    require('packer').startup {
        {
            'wbthomason/packer.nvim',
            'neovim/nvim-lspconfig',
        },
        config = {
            package_root = package_root,
            compile_path = compile_path,
        },
    }
end

_G.load_config = function()
    vim.lsp.set_log_level 'trace'
    if vim.fn.has 'nvim-0.5.1' == 1 then
        require('vim.lsp.log').set_format_func(vim.inspect)
    end
    local nvim_lsp = require 'lspconfig'
    local on_attach = function(_, bufnr)
        local function buf_set_option(...)
            vim.api.nvim_buf_set_option(bufnr, ...)
        end

        buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')

        -- Mappings.
        local opts = { buffer = bufnr, noremap = true, silent = true }
        vim.keymap.set('n', 'gD', vim.lsp.buf.declaration, opts)
        vim.keymap.set('n', 'gd', vim.lsp.buf.definition, opts)
        vim.keymap.set('n', 'K', vim.lsp.buf.hover, opts)
        vim.keymap.set('n', 'gi', vim.lsp.buf.implementation, opts)
        vim.keymap.set('n', '<C-k>', vim.lsp.buf.signature_help, opts)
        vim.keymap.set('n', '<space>wa', vim.lsp.buf.add_workspace_folder, opts)
        vim.keymap.set('n', '<space>wr', vim.lsp.buf.remove_workspace_folder, opts)
        vim.keymap.set('n', '<space>wl', function()
            print(vim.inspect(vim.lsp.buf.list_workspace_folders()))
        end, opts)
        vim.keymap.set('n', '<space>D', vim.lsp.buf.type_definition, opts)
        vim.keymap.set('n', '<space>rn', vim.lsp.buf.rename, opts)
        vim.keymap.set('n', 'gr', vim.lsp.buf.references, opts)
        vim.keymap.set('n', '<space>e', vim.diagnostic.open_float, opts)
        vim.keymap.set('n', '[d', vim.diagnostic.goto_prev, opts)
        vim.keymap.set('n', ']d', vim.diagnostic.goto_next, opts)
        vim.keymap.set('n', '<space>q', vim.diagnostic.setloclist, opts)
    end

    -- Add the server that troubles you here
    local name = 'denols'
    if not name then
        print 'You have not defined a server name, please edit minimal_init.lua'
    end
    if not nvim_lsp[name].document_config.default_config.cmd and not cmd then
        print [[You have not defined a server default cmd for a server
      that requires it please edit minimal_init.lua]]
    end

    nvim_lsp[name].setup {
        on_attach = on_attach,
        single_file_support = true,
    }

    print [[You can find your log at $HOME/.cache/nvim/lsp.log. Please paste in a github issue under a details tag as described in the issue template.]]
end

if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', 'https://github.com/wbthomason/packer.nvim', install_path }
    load_plugins()
    require('packer').sync()
    vim.cmd [[autocmd User PackerComplete ++once lua load_config()]]
else
    load_plugins()
    require('packer').sync()
    _G.load_config()
end
```

### LSP log

https://gist.github.com/sigmaSd/dbfd52efc9cf4f96e3e237f3cf65d9e8

**Comment by sigmaSd** [comment]
Commenting out this line https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/denols.lua#L33 appears to fix the issue.

**Comment by justinmk** [comment]
https://github.com/neovim/nvim-lspconfig/pull/1995

**Comment by sigmaSd** [comment]
This is not fixed by #1995 

In my testing it's fixed by commenting out this line https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/denols.lua#L38
